### ISSUE TITLE
Bug: DAX Query TOP1000 result limit not coherent with expected ORDER result

### ISSUE THREAD

**Issue by zenzeinet** [TabularEditor/TabularEditor3#913]
Hi, on TE370:

I write a DAX Query, lets say:

EVALUATE
ADDCOLUMNS(   VALUES( t[col]), "m", [metric])
ORDER BY [m] DESC

This Query returns many rows, lets say 1M rows, and I want to see the top t[col] by [metric]

TE3 automatically puts a TOPN(1000 around the query to limit the number of results. Good, but I think it is not well implemented:

What I suspect TE3 is doing:

EVALUATE
TOPN(1000 , ADDCOLUMNS(   VALUES( t[col]), "m", [metric]) )
ORDER BY [m] DESC

What it should be doing:

EVALUATE
TOPN(1000 , ADDCOLUMNS(   VALUES( t[col]), "m", [metric]) , [m], DESC) ---<<<<< TOPN SORT MISSING!
ORDER BY [m] DESC

I've found this because on real data, the above example was giving me low values of metric, when large ones should appear.
I tried the two formulas above with SSMS, and only the second gives proper results (that, or removing the topn alltogether)

**Comment by otykier** [comment]
Thanks for reporting this. I believe you're right, that TE3 doesn't apply the order explicitly specified to the TOPN that is used to reduce the result size.

As a workaround, until we get this fixed, write the query using a suitable TOPN, then click on the "click to get all rows" link at the bottom of the results pane (assuming your query returned more than 1000 rows):

<img width="521" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/8976200/94950c7b-23e9-410e-8b3b-511034f73259">

The row limit can also be configured under **Tools > Preferences > Data Browsing > DAX Query**:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/076d77e1-7715-4229-b626-d3429161118b)

**Comment by zenzeinet** [comment]
Hi,

Not sure if I saw it wrong or if it got fixed from 3.7.0 to 3.7.1 (to which I've upgraded recently). If I find it again I'll report back

Maybe I was doing something like

EVALUATE TOPN( 5000, Facts, Facts[col], desc )

In this case, if I don't add " ORDER BY Facts[col] DESC" at the end of the query, then the ( automatically adjusted topn 1000 result) result does not necessarily get the top values of Facts[col]. I understand why this is like this (works the same way in SQL - topN does not imply output sort order), although it can be a trap for users not aware of it. Maybe it could be nice if the intent was added to the adjusted topn 1000.

Then again I'm not sure that is whas I was doing in TE370, but I can't reproduce the bug now. (Note that the Evaluate topn example that I showed before does happen, but I would not consider it a "bug" although it can bite you if you're not careful)
